src/store/dataBaseLoadings.ts

```diff
@@ -0,0 +1,342 @@
+	    //const duration = date.getMinutes()+date.getHours()*60 - begin;
+        // console.log("Aktuelle Zeit   "+ Math.round(date.valueOf()/1000 ));
+        // console.log("Anfangszeit  "+ begin.seconds);
+        const duration = Math.round(date.valueOf()/1000 ) - begin.seconds;
+        sessions[sessions.length-1].duration = duration;
+        history[history.length-1].sessions = sessions;
+        const newAccumulateDuration = history[history.length-1].accumulated_duration + duration;
+        history[history.length-1].accumulated_duration = newAccumulateDuration;
+        const overAllDuration = docSnap.data().logs.overall_duration + duration;
+        await updateDoc(docRef, {
+            "logs.history": history,
+            "logs.overall_duration" : overAllDuration
+        });
+    } else {
+        // docSnap.data() will be undefined in this case
+        console.log("No such document!");
+    }
+
+
+}
+
+export const addNewHistory = async(docName: string, db: any, userData: DocumentData) =>{
+    const collectionsName = "users";
+    const docRef = doc(db, collectionsName, docName);
+    const dateObject = new Date();
+    const date = dateObject.toLocaleDateString();
+    const history = userData.logs.history;
+    console.log(date.localeCompare(history[history.length-1].date));
+    if (date.localeCompare(history[history.length-1].date)){
+        //current date is not included in history -> first login this day -> add history
+        
+        history.push(newHistoryArrayElement(dateObject));
+    }else{
+        //current date is already in log -> add only new session
+        const sessions = history[history.length-1].sessions;
+        //sessions.push(newSessionsArrayElement(dateObject));
+        history[history.length-1].sessions.push(newSessionsArrayElement(dateObject));
+    }
+    await updateDoc(docRef, {
+        "logs.history": history
+    });
+}
+
+export const newHistoryArrayElement = (date: Date )=> {
+    const newHistory = {
+        date: date.toLocaleDateString(),
+        accumulated_duration: 0,
+        sessions: [
+            {
+                //begin: (date.getMinutes()+date.getHours()*60),
+                begin: date,
+                duration: 0
+            }
+        ]
+     }
+     return newHistory;
+}
+
+export const newSessionsArrayElement = (date: Date )=> {
+    const newSession = {
+        //begin: (date.getMinutes()+date.getHours()*60),
+        begin: date,
+        duration:0
+     }
+     return newSession;
+}
+
+export const initDataStructure = (name: string, email: any )=> {
+    const data={
+        name: name,
+        email: email,
+        progress: {
+            overall_score: 0,
+            overall_questions:0,
+            genre_scores: {
+                classic: 0,
+                rock: 0,
+                pop: 0,
+                jazz: 0,
+                rap:0,
+                folk_music:0
+            },
+            music_period_scores: {
+                outdated: 0,
```

Comment:
For better readability, we could go with `"50s", "60s", etc`
```suggestion
                "50s": 0,
                "60s": 0,
```

Reply:
adjusted, we can speak about the exact categories we want to use for the music era in Sundays meeting